AirCasting/Utils/Bluetooth/NewBluetoothManager.swift

```diff
@@ -98,11 +98,17 @@ final class NewBluetoothManager: NSObject, BluetoothCommunicator, CBCentralManag
             lhs.peripheral == rhs.peripheral
         }
         
-        init(peripheral: CBPeripheral) {
+        fileprivate init(peripheral: CBPeripheral) {
             self.peripheral = peripheral
             name = peripheral.name
             uuid = peripheral.identifier.description
         }
+        
+        init(name: String?, uuid: String) {
+            self.name = name
+            self.uuid = uuid
+            peripheral = nil
+        }
```

Comment:
This is a codesmell (I know I'm the coauthor haha) indicating that we're lacking an abstraction here. What we would do is:
```Swift
// Use this in the rest of the codebase instead of the current `BluetoothDevice`
protocol BluetoothDevice: Equatable {
    var name: String? { get }
    var uuid: String { get }
}

// -------

// In production code (BluetoothManager+Device.swift):
extension BluetoothManager {
    private struct Device: BluetoothDevice {
        let name: String?
        let uuid: String
        let peripheral: CBPeripheral
    }
}

// -------

// In unit tests:
struct BluetoothDeviceMock: BluetoothDevice {
    var name: String?
    var uuid: String
}

```